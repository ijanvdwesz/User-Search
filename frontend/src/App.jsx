import React, { useState } from 'react'; // Imports React and the useState hook for managing component state
import UserSearch from './UserSearch.jsx'; // Imports the UserSearch component
import UserProfile from './UserProfile.jsx'; // Imports the UserProfile component
import './styles/App.css'; // Imports the external CSS file for styling

const App = () => {
  // State that stores the username entered by the user
  const [username, setUsername] = useState('');

  return (
    <div>
      {/* Main heading of the app */}
      <h1>GitHub User Search</h1>
      
      {/* Renders the UserSearch component and passes setUsername as a prop */}
      <UserSearch setUsername={setUsername} />
      
      {/* Renders the UserProfile component with the username,if a username is set */}
      {username && <UserProfile username={username} />}
    </div>
  );
};

export default App; // Exports the App component