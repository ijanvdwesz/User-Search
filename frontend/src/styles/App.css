/* Centers the text inside the main App container */
.App {
  text-align: center;
}

/* Styles the logo, setting its height and disabling pointer events */
.App-logo {
  height: 40vmin;
  pointer-events: none; /* Prevents interaction with the logo */
}

/* Adds a spinning animation to the logo when motion preference allows it */
@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

/* Styles the header with a dark background and centers content */
.App-header {
  background-color: #282c34; /* Dark background color */
  min-height: 100vh; /* Full viewport height */
  display: flex; /* Uses flexbox for alignment */
  flex-direction: column; /* Stacks children vertically */
  align-items: center; /* Centers items horizontally */
  justify-content: center; /* Centers items vertically */
  font-size: calc(10px + 2vmin); /* Responsive font size */
  color: white; /* White text color */
}

/* Defines the spinning animation */
@keyframes App-logo-spin {
  from {
    transform: rotate(0deg); /* Starts at 0-degree rotation */
  }
  to {
    transform: rotate(360deg); /* Rotates fully */
  }
}

/* Removes default styling for unordered lists */
ul {
  list-style-type: none; /* Removes bullet points */
  padding: 0; /* Removes default padding */
  margin: 0; /* Removes default margin */
}

/* Styles all anchor links */
a {
  color: rgb(7, 252, 7); /* Sets a bright green color */
}
