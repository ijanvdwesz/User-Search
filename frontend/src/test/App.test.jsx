// Imports necessary testing utilities and components
import { render, screen, fireEvent } from '@testing-library/react'; // From Testing Library that renders and interact with components
import App from '../App';  // Imports the App component to test

// Mocks the UserSearch and UserProfile components for testing
jest.mock('../UserSearch.jsx', () => () => <div>UserSearch Component</div>); // Mocks UserSearch to return a div
jest.mock('../UserProfile.jsx', () => () => <div>UserProfile Component</div>); // Mocks UserProfile to return a div

// Groups the tests related to the App component
describe('App Component', () => {
  
  // Tests to check if the app renders correctly with all necessary components
  it('renders the app with the search and profile components', () => {
    
    // Renders the App component
    render(<App />);
    
    // Assert that the title "GitHub User Search" is rendered
    expect(screen.getByText(/GitHub User Search/i)).toBeInTheDocument();
    
    // Assert that the mocked UserSearch component is rendered
    expect(screen.getByText(/UserSearch Component/i)).toBeInTheDocument();
    
    // Initially, the UserProfile component should not be rendered,(checks that it is not in the document)
    expect(screen.queryByText(/UserProfile Component/i)).toBeNull();
  });

  // Tests that checks if the app matches the snapshot (helps detect UI changes)
  it('matches the snapshot', () => {
    
    // Renders the App component
    const { asFragment } = render(<App />);
    
    // Compares the rendered output to the saved snapshot
    expect(asFragment()).toMatchSnapshot();
  });

});
