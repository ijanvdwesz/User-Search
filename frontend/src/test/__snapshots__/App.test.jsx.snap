// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component matches the snapshot 1`] = `
<DocumentFragment>
  <div>
    <h1>
      GitHub User Search
    </h1>
    <div>
      UserSearch Component
    </div>
  </div>
</DocumentFragment>
`;
